store/v2/migration/manager.go

```diff
@@ -183,13 +183,18 @@ func (m *Manager) writeChangeset() error {
 		batch := m.db.NewBatch()
 		// Invoking this code in a closure so that defer is called immediately on return
 		// yet not in the for-loop which can leave resource lingering.
-		err = func() error {
-			defer batch.Close()
+		err = func() (err error) {
+			defer func() {
+				cErr := batch.Close()
+				if err == nil {
+					err = cErr
+				}
+			}()
 
-			if err := batch.Set(csKey, csBytes); err != nil {
```

Comment:
You can revert the := for minimal changes. The way named returns work is that at return call site, the variable can have any name but the reference in the defer is what matters.